torch/distributed/_functional_collectives.py

```diff
@@ -395,12 +505,14 @@ def __tensor_unflatten__(inner_tensors, meta):
         return AsyncCollectiveTensor(elem)
 
     def __repr__(self):
-        wait_tensor(self.elem)
+        self.trigger_wait()
         return f"AsyncCollectiveTensor({self.elem})"
 
     def trigger_wait(self):
-        wait_tensor(self.elem)
-        return self
+        if not self.completed:
```

Comment:
I feel we should just delete the `trigger_wait` call, I don't actually quite see the value of this API instead of just calling `wait_tensor`?